### PR TITLE
fix: rtp -> vim.api.nvim_list_runtime_paths

### PR DESCRIPTION
**Pull request by goolord**
resolves https://github.com/shadmansaleh/lualine.nvim/issues/57

### CODE REVIEW

**Comment by dimbleby** [comment]
> Always use forward slashes (/) in the search pattern for subdirectories regardless of platform.

says the help.  So I think you want to lose the use of `sep` here.

It's less clear to me whether the returned paths also always use forward slashes, therefore whether `sep` should also be replaced by `'/'` [here](https://github.com/shadmansaleh/lualine.nvim/pull/59/files#diff-15c8cb3c84cb4afa68a2dad6ee86a4f1b6718c850e6d0d68865009d50e2127c1R238).

**Comment by goolord** [comment]
good catch,
the docs say
```
                Return: ~
                    list of absolute paths to the found files
```
i think 'absolute paths' implies platform dependent separators

**Comment by dimbleby** [comment]
> i think 'absolute paths' implies platform dependent separators

Seems a leap to me, but I'm in no position to test the Windows behaviour

**Comment by dimbleby** [comment]
Also you left `sep` in place in `loader.lua`, an oversight?

**Comment by goolord** [comment]
no, it's still used here https://github.com/shadmansaleh/lualine.nvim/pull/59/files#diff-901d2cca37037c403ccd996a740eebfb11cb0fedb808f96e18a002af34097781R26

i fail to see what else 'absolute path' could imply

**Comment by dimbleby** [comment]
In `loader.lua`:

>  local path = table.concat({ 'lua', 'lualine', 'themes', theme_name }, sep) .. '.lua'
>  local files = vim.api.nvim_get_runtime_file(path, true)

you left `sep`, but want a forward slash.

To my mind absolute path is completely orthogonal to the separator that is used: it means that it is not a relative path, and carries no implication at all about the separator.

**Comment by goolord** [comment]
what i mean is this: if this function is meant to resolve a search string to the absolute path of an *existing files* in your rtp, it follows to me that it is not going to return a filepath that contains invalid characters or one that is technically a valid absolute path but point to a completely different file than it found

**Comment by dimbleby** [comment]
You may be right but for what it's worth I would guess the other way: a function that insists on forward slashes (regardless of platform) in its input _probably_ also returns paths with forward slashes (regardless of platform).

But I think we are both guessing: absent documentation saying so one would have to test it (or perhaps read the source).

**Comment by goolord** [comment]
i read the source code 
https://github.com/neovim/neovim/blob/51a98aa0c2fe3231a0ffc8a78189bc6fafd6abf6/src/nvim/api/vim.c#L773
https://github.com/neovim/neovim/blob/master/src/nvim/runtime.c#L109-L140
the format of the `name` parameter is a pattern that is parsed by this piece of code, not a file path.
if you follow the execution of https://github.com/neovim/neovim/blob/master/src/nvim/runtime.c#L106 you can see that the path separator is platform dependent

**Comment by goolord** [comment]
i would be pretty surprised if the behavior you were speculating was the case, it would make this function useless on windows

**Comment by shadmansaleh** [comment]
Use `nvim_list_runtime_paths` inplace of splitting &rtp and leave the rest same .

both `nvim_list_runtime_paths` and `nvim_get_runtime_files` are available in 0.5 so it won't be a breaking change .

I didn't use `nvim_get_runtime_files` because it's slower does a lot more io operations than we need . Also don't like the fact it relies on shell for globing that's why rtp_searcher is there.

File: lua/lualine_require.lua [hunk 1]
```diff
@@ -45,7 +45,7 @@ end
 
 function M.rtp_searcher(file, once)
   local ret = {}
-  for dir in vim.gsplit(vim.api.nvim_get_option 'rtp', ',') do
+  for _, dir in pairs(vim.api.nvim_list_runtime_paths()) do
```
Comment (shadmansaleh) [inline lua/lualine_require.lua:48]: nitpick: `ipairs` is slightly faster than `pairs` . And since we are dealing with list it's better to use `ipairs`.

Comment (goolord) [reply lua/lualine_require.lua:48]: hm for some reason i was under the impression that this was not true of luajit. good to know

**Comment by shadmansaleh** [comment]
Thanks for fixing it :)

**Pull request merged**